File: src/Model/Statement/IfStatement.java
package Model.Statement;

import DataTypes.*;
import Model.Expression.Expression;
import Model.ProgramState;
import Exception.*;

/**
 * Created by vladc on 12.10.2016.
 */
public class IfStatement implements IStatement {
    private Expression expr;
    private IStatement thenStatement;
    private IStatement elseStatement;

    public IfStatement(Expression expr, IStatement thenStatement, IStatement elseStatement) {
        this.expr = expr;
        this.thenStatement = thenStatement;
        this.elseStatement = elseStatement;
    }

    @Override
    public String toString() {
        return "If (" + expr.toString() + ") Then\n\t(" +
                thenStatement.toString() + ")\nElse\n\t(" + elseStatement.toString() + ")";
    }

    public ProgramState execute(ProgramState state) throws MyStatementExecutionException {
        MyIStack<IStatement> st = state.getExeStack();
        MyIDictionary<String, Integer> symTab = state.getSymTable();
        MyIHeap heap = state.getHeap();

        try {
            if (expr.evaluate(symTab, heap) != 0)
                st.push(thenStatement);
            else
                st.push(elseStatement);
        }
        catch (ExpressionEvaluationException ex) {
            ex.printStackTrace();
        }
        return state;
    }
}

File: Toy-Language-Interpreter/Model/Statement/IfStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.Model.Expression;
using Toy_Language_Interpreter.DataTypes;
using Toy_Language_Interpreter.MyException;

namespace Toy_Language_Interpreter.Model.Statement
{
    class IfStatement : IStatement
    {
        private AExpression expr;
        private IStatement thenStatement;
        private IStatement elseStatement;

        public IfStatement(AExpression expr, IStatement thenStatement, IStatement elseStatement)
        {
            this.expr = expr;
            this.thenStatement = thenStatement;
            this.elseStatement = elseStatement;
        }

        public ProgramState Execute(ProgramState state)
        {
            MyIStack<IStatement> st = state.GetExeStack();
            MyIDictionary<string, int> symTab = state.GetSymTab();

            try
            {
                if (expr.Evaluate(symTab) != 0)
                    st.Push(thenStatement);
                else
                    st.Push(elseStatement);
            }
            catch (ExpressionEvaluationException e)
            {
                Console.Write(e.ToString());
            }
            return state;
        }

        public override string ToString()
        {
            return "If (" + expr.ToString() + ") Then (" +
                thenStatement.ToString() + ") Else (" + elseStatement.ToString() + ")";
        }
    }
}
